expression/builtin_compare_vec.go

```diff
@@ -590,11 +592,113 @@ func (b *builtinNullEQStringSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 }
 
 func (b *builtinLTIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinLTIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	var err error
+	var buf0, buf1 *chunk.Column
+	buf0, err = b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err = b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalString(b.ctx, input, buf1); err != nil {
+		return err
+	}
+
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf0, buf1)
+	vecCompareNull(buf0, buf1, result)
+	VecCompareInt(mysql.HasUnsignedFlag(b.args[0].GetType().Flag), mysql.HasUnsignedFlag(b.args[0].GetType().Flag), buf0, buf1, result)
+	vecResOfLT(result)
+	return nil
+}
+
+func vecResOfLT(result *chunk.Column) {
+	res := result.Int64s()
+	n := len(res)
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			res[i] = 0
+		}
+		if res[i] < 0 {
+			res[i] = 1
+		} else {
+			res[i] = 0
+		}
+	}
+}
+
+func vecCompareNull(largs, rargs, result *chunk.Column) {
```

Comment:
func vecCompareNull(lhs, rhs *chunk.Column, res []int64)

Reply:
I had modified it.

Reply:
I think the function "vecCompareNull" is useless, we may get rid of it。
Because in function "restOfLT", if one of the paramters is null, the result is 0.
![image](https://user-images.githubusercontent.com/13314548/66894882-19920680-f024-11e9-9482-f385e51c6831.png)

Please help to check whether it is correct. @XuHuaiyu

Reply:
when I try to get rid of "vecCompareNull", the unit test is correct.

Reply:
`vecCompareNull` is needed. (NullEQ)[https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#operator_equal-to] needs the result of it.

Reply:
Er... In the function "builtinLTIntSig" non-vectorized implementation:
when operands are null and null, the result is 0
when operands are 1 and null, the result is 0
when operands are null and 1, the result is 0

In the function "builtinLTIntSig" vectorized  implementation without using vecCompareNull:
when operands are null and null, the result is 0
when operands are 1 and null, the result is 0
when operands are null and 1, the result is 0

So, I think, in this function," vecCompareNull " is useless, maybe in other funciton, " vecCompareNull " is usefull.

In MySQL, only when the two operands are both null, it can be compared.
![image](https://user-images.githubusercontent.com/13314548/66901934-4cdb9200-f032-11e9-938f-5943597d2977.png)